speculation-rules/prerender/service-workers.https.html

```diff
@@ -0,0 +1,111 @@
+<script src="resources/utils.js"></script>
+<script src="/service-workers/service-worker/resources/test-helpers.sub.js"></script>
+<body>
+<script>
+
+setup(() => assertSpeculationRulesIsSupported());
+
+const ACTIVATION_DELAY = 500;
+
+promise_test(async t => {
+  const reg = await service_worker_unregister_and_register(
+    t, "./resources/service-worker.js", `resources/`);
+
+  t.add_cleanup(() => reg.unregister());
+  const {exec} = await create_prerendered_page(t);
+  const {text, prerendering} = await exec(async () => {
+    const text = await (await fetch(`ping`)).text();
+    return {text, prerendering: document.prerendering};
+  });
+
+  assert_true(prerendering);
+  assert_equals(text, 'pong');
+}, 'A prerendered page should be able to access an existing Service Worker');
+
+promise_test(async t => {
+  const {exec, activate} = await create_prerendered_page(t);
+  const scope = `/${token()}/`;
+  await exec(async scope => {
+    window.serviceWorkerInstalled = new Promise(resolve => {
+      navigator.serviceWorker.register('./service-worker.js', {scope: '.' + scope})
+        .then(reg => {
+          reg.unregister();
+          resolve({prerendering: document.prerendering});
+        });
+    });
+  }, scope);
+
+  await new Promise(resolve => t.step_timeout(resolve, ACTIVATION_DELAY));
+
+  await activate();
+
+  const {prerendering} = await exec(async () => { return await window.serviceWorkerInstalled});
+  assert_false(prerendering, 'Service Worker Installation should occur after activation');
+}, 'Registering a new service worker from a prerendered page should be delayed');
+
+promise_test(async t => {
+  const uid = token();
+  const reg = await service_worker_unregister_and_register(
+    t, "./resources/service-worker.js", `./resources/${uid}/`);
+  t.add_cleanup(() => reg.unregister());
+
+  const {exec, activate} = await create_prerendered_page(t);
+  await exec(async uid => {
+    window.serviceWorkerUpdate = new Promise(async resolve => {
+      const regs = await navigator.serviceWorker.getRegistrations();
+      const reg = regs.find(r => r.scope.includes(uid));
+      reg.update().then(() => resolve({prerendering: document.prerendering}));
+    });
+  }, uid);
+
+  await new Promise(resolve => t.step_timeout(resolve, ACTIVATION_DELAY));
+
+  await activate();
+
+  const {prerendering} = await exec(() => window.serviceWorkerUnregistered);
+  assert_false(prerendering, 'Service Worker deregistration should occur after activation');
+}, 'Unregistering an exsiting service worker from a prerendered page should be delayed');
+
+promise_test(async t => {
+  const uid = token();
+  const reg = await service_worker_unregister_and_register(
+    t, "./resources/service-worker.js", `./resources/${uid}/`);
+  t.add_cleanup(() => reg.unregister());
+
+  const {exec, activate} = await create_prerendered_page(t);
+  await exec(async uid => {
+    window.serviceWorkerUpdating = new Promise(async resolve => {
```

Comment:
nit: I guess it's a style preference, but I think this is marginally cleaner if we avoid mixing the Promise ctor and aync/await:

```javascript
window.serviceWorkerUpdating = (async () => {
  const regs = await navigator.serviceWorker.getRegistrations();
  const reg = regs.find(r => r.scope.includes(uid));
  await reg.update();
  return {prerendering: document.prerendering};
})();
```

Reply:
fixed